synapse/exc.py

```diff
@@ -125,6 +125,14 @@ class FileExists(SynErr):pass
 class NotEmpty(SynErr):pass
 class NotSupported(SynErr):pass
 
+class DatabaseInconsistent(SynErr):
```

Comment:
Maybe rename this BadCoreStore() to allow other storage layers to specify an error in this family without the terminology being specific to the idea of a database.